pkg/sloop/processing/eventcount.go

```diff
@@ -49,13 +48,37 @@ func updateEventCountTable(
 		return nil
 	}
 
-	// Truncate long-lived events to (watch.Timestamp - maxLookback).  This avoids filling in data that will immediately
+	// Truncate long-lived events to latest of (Now.Timestamp - maxLookBack - partitionDuration)
+	// and (Now.Timestamp - oldestPartitionTimeStamp - partitionDuration)
+	// This avoids filling in data that will immediately
 	// be garbage collected, and avoids creating transactions that are too large and fail
-	watchTs, err := ptypes.Timestamp(watchRec.Timestamp)
-	if err != nil {
-		return err
+
+	// Thus it solves the following scenario where new data is shown using *
+	// Before Event Count data is added:    MaxLookBack|----- Count: 50 ---|
+	// After Event Count data is added :    *****MaxLookBack|----- Count: 50 ---|
+
+	truncateTs := time.Now().UTC().Add(-1*maxLookback + untyped.GetPartitionDuration())
```

Comment:
I have big reservations about using `time.Now()`.  It makes for difficult unit tests.  And it means if we want to replay loading data later we will get different results.

The previous code used watchTs.  Is there a reason we can't stick with that?

Also, if we are querying the oldest partition do we even need this anymore?